### ISSUE TITLE
Folders not refreshed after some changes to folder through command line

### ISSUE THREAD

**Issue by ganeshkbhat** [Microsoft/vscode#15744]
- VSCode Version: 1.71.
- OS Version: Windows 10 Latest version

Steps to Reproduce:
Make a change in a folder - like create a file through command line. It sometimes or many times does not get refreshed automatically in the VSCode and I have to reopen the application to get the changes reflected. The reload option does not work during those times.

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by ganeshkbhat** [comment]
Here is the needed output from the console.

Overwriting extension <<.jsx>> to now point to mime <<text/x-jsx>>
mime.ts:59Overwriting extension <<.php>> to now point to mime <<text/x-hack>>
mime.ts:59Overwriting extension <<.hh>> to now point to mime <<text/x-hack>>
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  41004  for  code-runner.run
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  41000  for  code-runner.runByLanguage
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  41003  for  code-runner.stop
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  -1608343518  for  docthis.documentThis
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  -1608277982  for  docthis.documentEverything
keybindingsRegistry.ts:109Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  -1607163870  for  docthis.documentEverythingVisible
keybindingResolver.ts:190Conflict detected, command `workbench.action.debug.run` cannot be triggered by ctrl+f5 due to cordova.run

**Comment by BairDev** [comment]
I've seen the same problem, but in a different scenario. While working on a TypeScript/SystemJS/Angular2 project, the editor does not get some modules (intelli-sense) in the node_modules folder before running npm install. This is perfectly clear. But after running `npm install` VSCode apparently does not update the view in the editor, only after a restart the files in the new folder (node_modules) are recognized. This might also be fine, because of how the whole intelli-sense system works.

But I've seen a similar updating problem on e.g. doing some changes by using git in an external shell/console. In this case, the file in question in the editor did not get any updates, the view remained the same. Only a restart of VSCode could help.

Maybe it would be helpful to install a *update project/folder* feature?

Console output on running `npm install` (in a zsh-shell):

```
 code --verbose &
[1] 9436
[my-nema]@[my-machine] [my-project] (git)-[master] % [9456:1121/104141:ERROR:browser_main_loop.cc(231)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
[main 10:41:42 AM] Starting VS Code in verbose mode
[main 10:41:42 AM] from: /usr/share/code/resources/app
[main 10:41:42 AM] args: { _: [],
  help: false,
  h: false,
  version: false,
  v: false,
  wait: false,
  w: false,
  diff: false,
  d: false,
  goto: false,
  g: false,
  'new-window': false,
  n: false,
  'reuse-window': false,
  r: false,
  performance: false,
  p: false,
  verbose: true,
  logExtensionHostCommunication: false,
  'disable-extensions': false,
  disableExtensions: false,
  'list-extensions': false,
  'show-versions': false,
  nolazy: false }
[main 10:41:43 AM] IPC#vscode:startCrashReporter
[main 10:41:43 AM] IPC#vscode:setHeaders
[9456:1121/104144:INFO:CONSOLE(62)] "Conflict detected, command `editor.debug.action.toggleBreakpoint` cannot be triggered by f9 due to sortLines.sortLines", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (62)
[main 10:41:44 AM] IPC#vscode-workbenchLoaded
[9456:1121/104144:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104144:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104145:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104145:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[main 10:41:54 AM] Lifecycle#unload() 1
[main 10:41:55 AM] IPC#vscode:startCrashReporter
[main 10:41:55 AM] IPC#vscode:setHeaders
[main 10:41:55 AM] IPC#vscode-workbenchLoaded
[9456:1121/104156:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104156:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104156:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104156:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104156:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104157:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104215:INFO:CONSOLE(62)] "Conflict detected, command `editor.debug.action.toggleBreakpoint` cannot be triggered by f9 due to sortLines.sortLines", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (62)
[9456:1121/104226:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104226:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104301:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104302:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104303:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104304:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104304:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104304:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)
[9456:1121/104304:INFO:CONSOLE(34)] "%c[IPC Library: Watcher]", source: file:////usr/share/code/resources/app/out/vs/workbench/workbench.main.js (34)

```

Environment:

* Version 1.7.1
* Commit 02611b40b24c9df2726ad8b33f5ef5f67ac30b44
* Date 2016-11-03T13:46:46.244Z
* Shell 1.3.8
* Renderer 52.0.2743.82
* Node 6.5.0
* Linux Ubuntu (14.04) with Gnome
* Kernel: 4.4.0-47-generic

**Comment by bpasero** [comment]
@BairDev is this with having a folder opened in VS Code or single files that get modified outside using git?

**Comment by BairDev** [comment]
@bpasero The folder is opened in VSCode, in which the file lies, which get modified by git (e.g. using `git checkout path/to/file`). Even the file is opened in an editor in VSCode.

**Comment by bpasero** [comment]
@BairDev can you share the output of running with --verbose from the developer console inside VS Code (Help > Toggle Developer Tools > Console). It provides more info compared to the output on the command line.

**Comment by BairDev** [comment]
Sorry for the late response!

OK, `git checkout <file>` works.

    [IPC Library: Watcher] [CHANGED] /home/malte/workspace/glx-ubix-frontend_portal2/package.json
    ipc.cp.ts:141 [IPC Library: Watcher]  >> normalized [CHANGED] /home/malte/workspace/glx-ubix-frontend_portal2/.git/index
    ipc.cp.ts:141 [IPC Library: Watcher]  >> normalized [CHANGED] /home/malte/workspace/glx-ubix-frontend_portal2/package.json


And a `git rebase` also worked. I'm afraid that this was a race condition, so far I'm not able to reproduce it, although I remember, that it happened during some kind of git procedure (rebase etc.).

Sorry for that and I will open a new ticket, when I can reproduce the problem.

**Comment by bpasero** [comment]
Ok thanks.

**Comment by sricc** [comment]
I've been seeing this exact thing happen as well. I'm working on a project, remove a folder from the command line, but VSCode still shows that folder in the file explorer. The only thing I've found to work is to restart the application. I've definitely noticed it when using `npm link` while working on libraries within my projects. Within the library, which is symlinked via `npm link`, I remove a file or folder and it's still showing up in the editor.

VSCode: v1.7.2
macOS: 10.11.2

Let me know if there is anything I can do to help debug this.

**Comment by ganeshkbhat** [comment]
@bpasero sorry for the late response. I have this file opened in vscode. The changes are through either cmd prompt of windows.

**Comment by bpasero** [comment]
This is very likely https://github.com/Microsoft/vscode/issues/19377
